openassessment/staffgrader/staff_grader_mixin.py

```diff
@@ -140,13 +144,22 @@ def list_staff_workflows(self, data, suffix=''):  # pylint: disable=unused-argum
 
         # Fetch staff workflows, annotated with grading_status and lock_status
         staff_workflows = self._bulk_fetch_annotated_staff_workflows()
-        # Return seriaized staff workflows with additional Assessment / User / Team data.
-        # This is primarily split off in case we want to add pagination to this handler.
-        return self.staff_workflows_to_api_format(staff_workflows)
-
-    def staff_workflows_to_api_format(self, staff_workflows):
+        # Lookup additional info like usernames and assessments
+        serializer_context = self._get_list_workflows_serializer_context(staff_workflows)
+
+        # Serialize workflows with the context, and return the dict of submissions
+        result = {}
+        for staff_workflow in staff_workflows:
+            try:
+                serialized_workflow = SubmissionListSerializer(staff_workflow, context=serializer_context).data
+                result[staff_workflow.identifying_uuid] = serialized_workflow
+            except MissingContextException as e:
+                log.exception("Failed to serialize workflow %d: %s", staff_workflow.id, str(e), exc_info=True)
+        return result
```

Comment:
Obligatory, "should this live in a super serializer"?

Reply:
I don't think so because I don't know that there's a way to do variable keys through a serializer. If we wanted this to just be a list of submissions, yes, but like this, I don't think there's a way to do this that isn't messier than this.